### ISSUE TITLE
False positive on needless_continue when continuing to outer label

### ISSUE THREAD

**Issue by dtolnay** [rust-lang/rust-clippy#2329]
```rust
fn condition() -> bool { unimplemented!() }
fn update_condition() {}

fn main() {
    'outer: loop {
        println!("Entry");
        while condition() {
            update_condition();
            if condition() {
                // ...
            } else {
                continue 'outer;
            }
            println!("Hello, world");
        }
    }
}
```

Clippy suggests rewriting the `while` loop body to the following which is not correct.

```rust
update_condition();
if condition() {
    println!("Hello, world");
}
```
